crates/aranya-cli/src/main.rs

```diff
@@ -0,0 +1,1195 @@
+    }
+
+    fn store_stream(&mut self, stream: AqcBidiStream) -> String {
+        let id = Uuid::new_v4().to_string();
+        self.streams.insert(id.clone(), stream);
+        id
+    }
+
+    fn get_stream(&mut self, id: &str) -> Option<&mut AqcBidiStream> {
+        self.streams.get_mut(id)
+    }
+
+    fn store_peer_stream(&mut self, stream: AqcPeerStream) -> String {
+        let id = Uuid::new_v4().to_string();
+        self.peer_streams.insert(id.clone(), stream);
+        id
+    }
+
+    fn get_peer_stream(&mut self, id: &str) -> Option<&mut AqcPeerStream> {
+        self.peer_streams.get_mut(id)
+    }
+
+    fn store_received_channel(&mut self, channel: AqcPeerChannel) -> String {
+        let id = Uuid::new_v4().to_string();
+        self.received_channels.insert(id.clone(), channel);
+        id
+    }
+
+    fn get_received_channel(&mut self, id: &str) -> Option<&mut AqcPeerChannel> {
+        self.received_channels.get_mut(id)
+    }
+
+    fn list_channels(&self) -> Vec<String> {
+        self.channels.keys().cloned().collect()
+    }
+
+    fn list_streams(&self) -> Vec<String> {
+        self.streams.keys().cloned().collect()
+    }
+
+    fn list_peer_streams(&self) -> Vec<String> {
+        self.peer_streams.keys().cloned().collect()
+    }
+
+    fn list_received_channels(&self) -> Vec<String> {
+        self.received_channels.keys().cloned().collect()
+    }
+
+    fn remove_channel(&mut self, id: &str) -> Option<AqcBidiChannel> {
+        self.channels.remove(id)
+    }
+
+    fn remove_stream(&mut self, id: &str) -> Option<AqcBidiStream> {
+        self.streams.remove(id)
+    }
+
+    fn remove_peer_stream(&mut self, id: &str) -> Option<AqcPeerStream> {
+        self.peer_streams.remove(id)
+    }
+
+    fn remove_received_channel(&mut self, id: &str) -> Option<AqcPeerChannel> {
+        self.received_channels.remove(id)
+    }
+}
+
+// Global registry instance
+lazy_static! {
+    static ref CHANNEL_REGISTRY: Arc<Mutex<ChannelRegistry>> = Arc::new(Mutex::new(ChannelRegistry::new()));
+}
+
+#[derive(Parser)]
+#[command(name = "aranya")]
+#[command(author, version, about = "Aranya CLI tool for team and device management", long_about = None)]
+struct Cli {
+    /// Path to daemon's Unix Domain Socket
+    #[arg(short = 'u', long, default_value = "/var/run/aranya/uds.sock")]
+    uds_path: PathBuf,
+
+    /// Daemon's AQC address
+    #[arg(short = 'a', long, default_value = "127.0.0.1:0")]
+    aqc_addr: String,
+
+    /// Enable verbose logging
+    #[arg(short, long)]
+    verbose: bool,
+
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Create a new team
+    CreateTeam {
+        /// Optional seed IKM in hex (32 bytes). If not provided, generates random.
+        #[arg(long)]
+        seed_ikm: Option<String>,
+    },
+    /// Add an existing team to this device
+    AddTeam {
+        /// Team ID to add
+        team_id: String,
```

Comment:
It would be better to define actual ID types here.
```suggestion
        team_id: TeamId,
```
If this doesn't compile, it's possible there are impls that need to be implemented to parse the String from the CLI arg into a specific type.